include/cantera/thermo/PengRobinson.h

```diff
@@ -0,0 +1,361 @@
+     *
+     * @param ac Output vector of activity coefficients. Length: m_kk.
+     */
+    virtual void getActivityCoefficients(double* ac) const;
+
+    /// @name  Partial Molar Properties of the Solution
+    //@{
+
+    //! Get the array of non-dimensional species chemical potentials.
+    //! These are partial molar Gibbs free energies.
+    /*!
+     * \f$ \mu_k / \hat R T \f$.
+     * Units: unitless
+     *
+     *
+     * @param mu    Output vector of non-dimensional species chemical potentials
+     *              Length: m_kk.
+     */
+    
+    virtual void getChemPotentials(double* mu) const;
+    virtual void getPartialMolarEnthalpies(double* hbar) const;
+    virtual void getPartialMolarEntropies(double* sbar) const;
+    virtual void getPartialMolarIntEnergies(double* ubar) const;
+    virtual void getPartialMolarCp(double* cpbar) const;
+    virtual void getPartialMolarVolumes(double* vbar) const;
+
+    //! Calculate the temperature dependent interaction parameter \f$\alpha\f$ needed for P-R EoS
+    /*!
+     *  The temperature dependent parameter in P-R EoS is calculated as
+     *       \f$ \alpha = [1 + \kappa(1 - \sqrt{T/T_{crit}})]^2  \f$
+     *  kappa is a function calulated based on the acentric factor.
+     *  Units: unitless
+     * 
+     * @param a    species-specific coefficients used in P-R EoS
+     * @param b    species-specific coefficients used in P-R EoS
+     * @param w    the acentric factor
+     */
+
+    virtual void calculateAlpha(const std::string& species, double a, double b, double w);
+    //@}
+    /// @name Critical State Properties.
+    //@{
+
+    virtual double critTemperature() const;
+    virtual double critPressure() const;
+    virtual double critVolume() const;
+    virtual double critCompressibility() const;
+    virtual double critDensity() const;
+    virtual double speciesCritTemperature(double a, double b) const;
+
+public:
+    //@}
+    //! @name Initialization Methods - For Internal use
+    /*!
+     * The following methods are used in the process of constructing
+     * the phase and setting its parameters from a specification in an
+     * input file. They are not normally used in application programs.
+     * To see how they are used, see importPhase().
+     */
+    //@{
+
+    virtual bool addSpecies(shared_ptr<Species> spec);
+    virtual void initThermo();
+
+    //! Retrieve a and b coefficients by looking up tabulated critical parameters
+    /*!
+    *  If pureFluidParameters are not provided for any species in the phase,
+    *  consult the critical properties tabulated in data/inputs/critProperties.xml.
+    *  If the species is found there, calculate pure fluid parameters a_k and b_k as:
+    *  \f[ a_k = 0.4278 R^2 T_c^2 / P_c \f]
+    *
+    *  and:
+    *  \f[ b_k = 0.08664 R T_c/ P_c \f]
+    *
+    *  @param iName    Name of the species
+    */
+    virtual std::vector<double> getCoeff(const std::string& iName);
```

Comment:
```suggestion
    virtual vector_fp getCoeff(const std::string& iName);
```

Also, should this be a `protected` method? It seems like more of an implementation detail than something a user of the class would want access to.